### PR TITLE
Support for BigQuery's NUMERIC type, which is currently in alpha.

### PR DESCRIPTION
**Pull request by ElliottBrossard**

### CODE REVIEW

**Comment by tswast** [comment]
Strange that googlebot is having trouble with the CLA. I see you listed at go/github. Maybe you used the wrong email on the git commit author metadata?

**Comment by ElliottBrossard** [comment]
I fixed the email address on the commit. Maybe googlebot will pick up the change now.

File: bigquery/tests/system.py [hunk 1]
```diff
@@ -838,6 +839,10 @@ def _generate_standard_sql_types_examples(self):
                 'sql': 'SELECT TIME(TIMESTAMP "%s")' % (stamp,),
                 'expected': naive.time(),
             },
+            {
+                'sql': 'SELECT NUMERIC "123456789.123456789"',
+                'expected': decimal.Decimal("123456789.123456789"),
```
Comment (tswast) [inline bigquery/tests/system.py]: nit: use single quotes for strings

File: bigquery/tests/system.py [hunk 2]
```diff
@@ -999,6 +1004,10 @@ def test_query_w_query_params(self):
         pi = 3.1415926
         pi_param = ScalarQueryParameter(
             name='pi', type_='FLOAT64', value=pi)
+        pi_numeric = decimal.Decimal("3.141592654")
```
Comment (tswast) [inline bigquery/tests/system.py]: nit: use single quotes for strings

File: bigquery/tests/system.py (hunk 1 above)
Comment (ElliottBrossard) [reply bigquery/tests/system.py]: Done.

File: bigquery/tests/system.py (hunk 2 above)
Comment (ElliottBrossard) [reply bigquery/tests/system.py]: Done.

**Comment by tswast** [comment]
This repository requires ~100% code coverage for unit tests. Circle CI is failing due to missing coverage on

```
google/cloud/bigquery/_helpers.py             185      2     83      0    99%   52-53
google/cloud/bigquery/dbapi/_helpers.py        39      1     26      1    97%   51, 50->51
```

**Comment by tswast** [comment]
Oh, and you should rebase to get the flake8 fixes for `snippets.py`.

**Comment by ElliottBrossard** [comment]
Okay, I think this is ready for review again. Sorry for the dumb question, but...what are the flake8 fixes for snippets.py? Is there some additional test that I should be running?

**Comment by tswast** [comment]
Flake8 the linting tool we use (for enforcing pep-8 style). If you follow the instructions to install nox at https://github.com/GoogleCloudPlatform/google-cloud-python/blob/master/CONTRIBUTING.rst and then run `nox -s lint` it will run the linter in the same way that circleci is configured. Running just `nox` will run all tests and lint on multiple versions of Python.

**Comment by chemelnucfin** [comment]
@ElliottBrossard Hi, it is probably best if you used `git pull rebase --upstream master` if you set your upstream repo to be `GoogleCloudPlatform/google-cloud-python`.  That way you will rewind and put all of your work at the end of the chain.

**Comment by chemelnucfin** [comment]
you can also try `git rebase -i` and go back however many commits you need to straigten out the commit list.

**Comment by ElliottBrossard** [comment]
I think we're in a better state--tswast updated the bigquery-numeric branch and the commit history for this PR is clean.

**Pull request merged**